python/lsst/ts/athexapod/csc.py

```diff
@@ -481,38 +489,36 @@ async def telemetry(self):
 
         sub_tasks = 2
         while self.run_telemetry_task:
-            # Get setpointPosition
-            target_position = await self.controller.target_position()
+            try:
+                # Get setpointPosition
+                target_position = await self.controller.target_position()
 
-            # Get reportedPosition
-            current_position = await self.controller.real_position()
+                # Get reportedPosition
+                current_position = await self.controller.real_position()
 
-            diff = [current_position[i] - target_position[i] for i in range(6)]
+                diff = [current_position[i] - target_position[i] for i in range(6)]
 
-            await self.tel_positionStatus.set_write(
-                setpointPosition=target_position,
-                reportedPosition=current_position,
-                positionFollowingError=diff,
-            )
+                await self.tel_positionStatus.set_write(
+                    setpointPosition=target_position,
+                    reportedPosition=current_position,
+                    positionFollowingError=diff,
+                )
 
-            await asyncio.sleep(self.heartbeat_interval / sub_tasks)
+                await asyncio.sleep(self.heartbeat_interval / sub_tasks)
 
-            # Check for errors
-            error = await self.controller.get_error()
-            if error != 0:
+                # Check for errors
+                error = await self.controller.get_error()
+                if error != 0:
+                    await self.fault(code=error, report=translate_error(error))
+                else:
+                    await asyncio.sleep(self.heartbeat_interval / sub_tasks)
+            except asyncio.CancelledError:
+                self.log.info("Telemetry task cancelled.")
+            except RuntimeError:
+                self.log.exception("ATHexapod unexpectedly disconnected.")
                 await self.fault(
-                    code=error, report=translate_error(error), traceback=""
+                    code=100000, report="ATHexapod unexpectedly disconnected."
```

Comment:
Why are you changing this?

Reply:
I'm adding that fault transition. It's for if the ATHexapod is unexpectedly disconnected. The error fault transition is in 511-512